### PR TITLE
161 proposal for new bufr table entries for eps sg eumetsat iasi ng level 2 and vii level 2

### PR DESCRIPTION
**Pull request by amilan17**
#161

### CODE REVIEW

**Comment by antoinemerle** [comment]
Thanks @amilan17.

**Comment by amilan17** [comment]
@antoinemerle I made some editorial changes (see [https://github.com/wmo-im/BUFR4/compare/de70ce8e56c2f2a5fee11be858b5a9860c398521...161-proposal-for-new-bufr-table-entries-for-eps-sg-eumetsat-iasi-ng-level-2-and-vii-level-2)](https://github.com/wmo-im/BUFR4/compare/de70ce8e56c2f2a5fee11be858b5a9860c398521...161-proposal-for-new-bufr-table-entries-for-eps-sg-eumetsat-iasi-ng-level-2-and-vii-level-2)) and have some comments for your consideration. 

Thanks.

**Comment by antoinemerle** [comment]
Hi @amilan17 ,

Quick question : When you mean "Have comments for your consideration" you mean that probably I "click" on the Ready for review ? I am not able to see any of your comments apart the changes you've made

Second point which is interesting for my future self , I got an automated scripts that push the changes inside the PR branch. I would be interested that you confirm the two follow assumption :
- table should always start with capital 'T' : "Table"
- the following char "-" is prohibited in code name 

thanks a million for confirming that, I will update my script accordingly :)

**Comment by amilan17** [comment]
@antoinemerle 

> Quick question : When you mean "Have comments for your consideration" you mean that probably I "click" on the Ready for review ? I am not able to see any of your comments apart the changes you've made

I forgot to submit my review -- you should see the comments now. 

> *   table should always start with capital 'T' : "Table"

T should not be capitalized when referencing "Code table" and "Flag table"

> *   the following char "-" is prohibited in code name

No, it's not prohibited. I removed it in "Water-vapour" for consistency with same term in other tables

File: BUFR_TableD_en_10.csv [hunk 1]
```diff
@@ -1854,6 +1854,262 @@ Category,CategoryOfSequences_en,FXY1,Title_en,SubTitle_en,FXY2,ElementName_en,El
...
+10,Vertical sounding sequences (satellite data),310088,"(Metop-SG Infrared Atmospheric Sounder Interferometer - New Generation (IASI-NG), level 2 temperature and water vapour profiles (TWV))",,013001,Specific humidity,Humidity at given pressure,,,Operational
+10,Vertical sounding sequences (satellite data),310089,"(Metop-SG Infrared Atmospheric Sounder Interferometer - New Generation (IASI-NG), level 2 cloud detection (CLD))",,310087,IASI NG L2 common information part,"The common information shared for TWV, CLD, SFC and GHG",,,Operational
+10,Vertical sounding sequences (satellite data),310089,"(Metop-SG Infrared Atmospheric Sounder Interferometer - New Generation (IASI-NG), level 2 cloud detection (CLD))",,012083,Temperature difference between observed and computed,Temperature difference in Kelvin between the observed temperature and the computed temperature,,,Operational
+10,Vertical sounding sequences (satellite data),310089,"(Metop-SG Infrared Atmospheric Sounder Interferometer - New Generation (IASI-NG), level 2 cloud detection (CLD))",,008043,Atmospheric chemical or physical constituent type,Set to Dust,,,Operational
```
[the remark this answers is not in the document]
Comment (amilan17) [reply BUFR_TableD_en_10.csv:1936]: use 30 Crustal material ?

File: BUFR_TableD_en_10.csv [hunk 2]
```diff
@@ -1854,6 +1854,262 @@ Category,CategoryOfSequences_en,FXY1,Title_en,SubTitle_en,FXY2,ElementName_en,El
...
+10,Vertical sounding sequences (satellite data),310090,"(Metop-SG Infrared Atmospheric Sounder Interferometer - New Generation (IASI-NG), level 2 surface parameters (SFC))",,040044,Dust index,,,,Operational
+10,Vertical sounding sequences (satellite data),310090,"(Metop-SG Infrared Atmospheric Sounder Interferometer - New Generation (IASI-NG), level 2 surface parameters (SFC))",,008043,Atmospheric chemical or physical constituent type,Cancel Dust,,,Operational
+10,Vertical sounding sequences (satellite data),310091,"(Metop-SG Infrared Atmospheric Sounder Interferometer - New Generation (IASI-NG), level 2 total columns of nitrous oxide, methane and carbon dioxide (GHG))",,310087,IASI NG L2 common information part,"The common information shared for TWV, CLD, SFC and GHG",,,Operational
+10,Vertical sounding sequences (satellite data),310091,"(Metop-SG Infrared Atmospheric Sounder Interferometer - New Generation (IASI-NG), level 2 total columns of nitrous oxide, methane and carbon dioxide (GHG))",,104003,Replicate 4 descriptors 3 times ,Loop other the 3 different chemical constituent type,,,Operational
```
[the remark this answers is not in the document]
Comment (amilan17) [reply BUFR_TableD_en_10.csv]: Repeat for the 3 different chemical ...

File: BUFRCREX_TableB_en_20.csv [hunk 3]
```diff
@@ -108,3 +108,4 @@ ClassNo,ClassName_en,FXY,ElementName_en,BUFR_Unit,BUFR_Scale,BUFR_ReferenceValue
 20,Observed phenomena,020136,Supplementary cloud type,Code table,0,0,9,Code table,0,3,,,Operational
 20,Observed phenomena,020137,Evolution of clouds,Code table,0,0,4,Code table,0,2,,,Operational
 20,Observed phenomena,020138,Road surface condition,Code table,0,0,4,Code table,0,2,,,Operational
+20,Observed phenomena,020139,Final cloud model assumption,Code table,0,0,4,Code table,0,2,,,Operational
```
[the remark this answers is not in the document]
Comment (antoinemerle) [reply BUFRCREX_TableB_en_20.csv:111]: Hi @amilan17 :  you're completely correct here :

✔️ Correct name is : `Final cloud model assumption`
📛 Bad is : `Final cloud model assumption for VII L2`

File: BUFR_TableD_en_10.csv (hunk 1 above)
[the remark this answers is not in the document]
Comment (antoinemerle) [reply BUFR_TableD_en_10.csv:1936]: Hi @amilan17 : Correct, we are using `30` in our sample data that is corresponding to `Crustal material (Dust)`

File: BUFRCREX_TableB_en_20.csv (hunk 3 above)
[the remark this answers is not in the document]
Comment (antoinemerle) [reply BUFRCREX_TableB_en_20.csv:111]: - [ ] update in BUFRCREX_CodeFlag_en_20.csv

File: BUFR_TableD_en_10.csv (hunk 1 above)
[the remark this answers is not in the document]
Comment (antoinemerle) [reply BUFR_TableD_en_10.csv:1936]: - [ ] update and use the smae word as in the table B

File: BUFR_TableD_en_10.csv (hunk 2 above)
[the remark this answers is not in the document]
Comment (antoinemerle) [reply BUFR_TableD_en_10.csv]: - [ ] change from "loop the other 3 differents.." to "Repeat for the 3 different chemical..."

File: BUFR_TableD_en_10.csv [hunk 4]
```diff
@@ -1854,6 +1854,262 @@ Category,CategoryOfSequences_en,FXY1,Title_en,SubTitle_en,FXY2,ElementName_en,El
...
+10,Vertical sounding sequences (satellite data),310092,"(Metop-SG Infrared Atmospheric Sounder Interferometer - New Generation (IASI-NG), level 2 common information part (IASI NG L2 CO)",,005021,Bearing or azimuth,Measurement azimuth angle at sounder pixel centre,,,Operational
+10,Vertical sounding sequences (satellite data),310092,"(Metop-SG Infrared Atmospheric Sounder Interferometer - New Generation (IASI-NG), level 2 common information part (IASI NG L2 CO)",,007025,Solar zenith angle,Solar zenith angle at sounder pixel centre,,,Operational
+10,Vertical sounding sequences (satellite data),310092,"(Metop-SG Infrared Atmospheric Sounder Interferometer - New Generation (IASI-NG), level 2 common information part (IASI NG L2 CO)",,005022,Solar azimuth,Solar azimuth angle at sounder pixel centre,,,Operational
+10,Vertical sounding sequences (satellite data),310092,"(Metop-SG Infrared Atmospheric Sounder Interferometer - New Generation (IASI-NG), level 2 common information part (IASI NG L2 CO)",,005043,Field of view number,Field of view number,,,Operational
```
[the remark this answers is not in the document]
Comment (antoinemerle) [reply BUFR_TableD_en_10.csv:2000]: - [ ] : remove the redundant information that is already mentionned in its default name

File: BUFR_TableD_en_10.csv [hunk 5]
```diff
@@ -1854,6 +1854,262 @@ Category,CategoryOfSequences_en,FXY1,Title_en,SubTitle_en,FXY2,ElementName_en,El
...
+10,Vertical sounding sequences (satellite data),310092,"(Metop-SG Infrared Atmospheric Sounder Interferometer - New Generation (IASI-NG), level 2 common information part (IASI NG L2 CO)",,005040,Orbit number,Orbit number,,,Operational
```
[the remark this answers is not in the document]
Comment (antoinemerle) [reply BUFR_TableD_en_10.csv]: - [ ] : remove the redundant information that is already mentionned in its default name

File: BUFR_TableD_en_10.csv [hunk 6]
```diff
@@ -1854,6 +1854,262 @@ Category,CategoryOfSequences_en,FXY1,Title_en,SubTitle_en,FXY2,ElementName_en,El
...
+10,Vertical sounding sequences (satellite data),310094,"(Metop-SG Infrared Atmospheric Sounder Interferometer - New Generation (IASI-NG), level 2 nitric acid total column, nitric acid profile (NAC) and level 2 ozone total column, ozone profile (O3)",,040062,A-priori partial columns for each retrieved layer,A-priori partial columns for HNO3 or O3 in each retrieved layer,,,Operational
+10,Vertical sounding sequences (satellite data),310094,"(Metop-SG Infrared Atmospheric Sounder Interferometer - New Generation (IASI-NG), level 2 nitric acid total column, nitric acid profile (NAC) and level 2 ozone total column, ozone profile (O3)",,040063,Scaling vector multiplying the a priori vector in order to define the retrieved vector,Scaling vector multiplying the a priori HNO3 or O3 vector in order to define the retrieved HNO3 or O3 vector.,,,Operational
+10,Vertical sounding sequences (satellite data),310094,"(Metop-SG Infrared Atmospheric Sounder Interferometer - New Generation (IASI-NG), level 2 nitric acid total column, nitric acid profile (NAC) and level 2 ozone total column, ozone profile (O3)",,102021,Repeat 2 descriptor 21 times,Layer repeat 861,,,Operational
```
[the remark this answers is not in the document]
Comment (antoinemerle) [reply BUFR_TableD_en_10.csv:2034]: - [ ] : be more consitent in the description like ;
- [ ] : additionnal comment at the same time :   | 103041 | Repeat 3 descriptor 41 times | Loop other each layer

File: BUFR_TableD_en_10.csv [hunk 7]
```diff
@@ -1854,6 +1854,262 @@ Category,CategoryOfSequences_en,FXY1,Title_en,SubTitle_en,FXY2,ElementName_en,El
...
+10,Vertical sounding sequences (satellite data),310096,"(Metop-SG Visible Infrared Imager (VII), level 2 common part for VII L2 CLD, CTP, OCA, WVI, WVV)",,007024,Satellite zenith angle,,,,Operational
+10,Vertical sounding sequences (satellite data),310096,"(Metop-SG Visible Infrared Imager (VII), level 2 common part for VII L2 CLD, CTP, OCA, WVI, WVV)",,005021,Bearing or azimuth (degree true),,,,Operational
+10,Vertical sounding sequences (satellite data),310096,"(Metop-SG Visible Infrared Imager (VII), level 2 common part for VII L2 CLD, CTP, OCA, WVI, WVV)",,007025,Solar zenith angle,,,,Operational
+10,Vertical sounding sequences (satellite data),310096,"(Metop-SG Visible Infrared Imager (VII), level 2 common part for VII L2 CLD, CTP, OCA, WVI, WVV)",,005022,Solar azimuth (degree true),,,,Operational
+10,Vertical sounding sequences (satellite data),310096,"(Metop-SG Visible Infrared Imager (VII), level 2 common part for VII L2 CLD, CTP, OCA, WVI, WVV)",,033028,Snapshot overall quality,"Processing condition indicator (0=nominal, 1=General degradation)",,,Operational
```
[the remark this answers is not in the document]
Comment (antoinemerle) [reply BUFR_TableD_en_10.csv:2063]: - [ ] : in desc : change `1=General degradation` to `5=General degradation`

File: BUFR_TableD_en_10.csv [hunk 8]
```diff
@@ -1854,6 +1854,262 @@ Category,CategoryOfSequences_en,FXY1,Title_en,SubTitle_en,FXY2,ElementName_en,El
...
+10,Vertical sounding sequences (satellite data),310097,"(Metop-SG Visible Infrared Imager (VII), level 2 Cloud Mask, Cloud Detection (CLD))",,310096,VII L2 common information part,"The common information shared for CLD, CTP, OCA, WVV and WVI",,,Operational
+10,Vertical sounding sequences (satellite data),310097,"(Metop-SG Visible Infrared Imager (VII), level 2 Cloud Mask, Cloud Detection (CLD))",,008011,Meteorological feature,Set to 11 to specify we are speaking about cloud clear,,,Operational
+10,Vertical sounding sequences (satellite data),310097,"(Metop-SG Visible Infrared Imager (VII), level 2 Cloud Mask, Cloud Detection (CLD))",,033007,Per cent confidence,Pixel clear-sky confidence,,,Operational
+10,Vertical sounding sequences (satellite data),310097,"(Metop-SG Visible Infrared Imager (VII), level 2 Cloud Mask, Cloud Detection (CLD))",,008011,Meteorological feature,Set to cancel,,,Operational
+10,Vertical sounding sequences (satellite data),310097,"(Metop-SG Visible Infrared Imager (VII), level 2 Cloud Mask, Cloud Detection (CLD))",,008072,Pixel type,Cloudy flag,,,Operational
```
[the remark this answers is not in the document]
Comment (amilan17) [reply BUFR_TableD_en_10.csv:2068]: \=2 Cloudy

[the remark this answers is not in the document]
Comment (antoinemerle) [reply BUFR_TableD_en_10.csv:2068]: - [] : set to ` set to Cloudy`

File: BUFR_TableD_en_10.csv [hunk 9]
```diff
@@ -1854,6 +1854,262 @@ Category,CategoryOfSequences_en,FXY1,Title_en,SubTitle_en,FXY2,ElementName_en,El
...
+10,Vertical sounding sequences (satellite data),310099,"(Metop-SG Visible Infrared Imager (VII), level 2 Optimal Cloud Analysis (OCA))",,310096,VII L2 common information part,"The common information shared for CLD, CTP, OCA, WVV and WVI",,,Operational
+10,Vertical sounding sequences (satellite data),310099,"(Metop-SG Visible Infrared Imager (VII), level 2 Optimal Cloud Analysis (OCA))",,025144,Number of visual channels used in processing,From 0 to 11,,,Operational
+10,Vertical sounding sequences (satellite data),310099,"(Metop-SG Visible Infrared Imager (VII), level 2 Optimal Cloud Analysis (OCA))",,025145,Number of infrared channels used in processing,From 5 to 9,,,Operational
+10,Vertical sounding sequences (satellite data),310099,"(Metop-SG Visible Infrared Imager (VII), level 2 Optimal Cloud Analysis (OCA))",,020139,Final cloud model assumption,"Final cloud model assumption: 0=clear ,1=single layer water,2=single layer ice,3=2-layers (water/ice),4=single layer ash,5=2-layers ash",,,Operational
```
[the remark this answers is not in the document]
Comment (antoinemerle) [reply BUFR_TableD_en_10.csv]: - [ ] : correct : overwhelming information

File: BUFR_TableD_en_10.csv [hunk 10]
```diff
@@ -1854,6 +1854,262 @@ Category,CategoryOfSequences_en,FXY1,Title_en,SubTitle_en,FXY2,ElementName_en,El
...
+10,Vertical sounding sequences (satellite data),310087,"(Metop-SG Infrared Atmospheric Sounder Interferometer - New Generation (IASI-NG), level 2 common information part (TWV, CLD, SFC, GHG))",,040047,Validation flag for IASI or IASI-NG level 1 product,,,,Operational
+10,Vertical sounding sequences (satellite data),310087,"(Metop-SG Infrared Atmospheric Sounder Interferometer - New Generation (IASI-NG), level 2 common information part (TWV, CLD, SFC, GHG))",,008065,Sun-glint indicator,,,,Operational
+10,Vertical sounding sequences (satellite data),310087,"(Metop-SG Infrared Atmospheric Sounder Interferometer - New Generation (IASI-NG), level 2 common information part (TWV, CLD, SFC, GHG))",,204004,Add associated field with 4 bits,Adding 4 bits to give quality information on the data to be processed,,,Operational
+10,Vertical sounding sequences (satellite data),310087,"(Metop-SG Infrared Atmospheric Sounder Interferometer - New Generation (IASI-NG), level 2 common information part (TWV, CLD, SFC, GHG))",,031021,Associated field significance,9 for specifying for each of the incoming variable the status of the data retrieved.,,,Operational
```
[the remark this answers is not in the document]
Comment (antoinemerle) [reply BUFR_TableD_en_10.csv]: - [ ] set to ` set to 9 to provide information for the following descriptor`

File: BUFR_TableD_en_10.csv (hunk 2 above)
[the remark this answers is not in the document]
Comment (antoinemerle) [reply BUFR_TableD_en_10.csv]: I also remove all the places I were using "loop"

File: BUFR_TableD_en_10.csv [hunk 11]
```diff
@@ -1854,6 +1854,262 @@ Category,CategoryOfSequences_en,FXY1,Title_en,SubTitle_en,FXY2,ElementName_en,El
...
+10,Vertical sounding sequences (satellite data),310090,"(Metop-SG Infrared Atmospheric Sounder Interferometer - New Generation (IASI-NG), level 2 surface parameters (SFC))",,008003,Vertical significance (satellite observations),Cancel Vertical Significance,,,Operational
+10,Vertical sounding sequences (satellite data),310090,"(Metop-SG Infrared Atmospheric Sounder Interferometer - New Generation (IASI-NG), level 2 surface parameters (SFC))",,012083,Temperature difference between observed and computed,Difference between the observed temperature and the computed temperature,,,Operational
+10,Vertical sounding sequences (satellite data),310090,"(Metop-SG Infrared Atmospheric Sounder Interferometer - New Generation (IASI-NG), level 2 surface parameters (SFC))",,011012,Wind speed at 10m,,,,Operational
+10,Vertical sounding sequences (satellite data),310090,"(Metop-SG Infrared Atmospheric Sounder Interferometer - New Generation (IASI-NG), level 2 surface parameters (SFC))",,008043,Atmospheric chemical or physical constituent type,Set Atmospheric chemical to Dust,,,Operational
```
[the remark this answers is not in the document]
Comment (antoinemerle) [reply BUFR_TableD_en_10.csv]: Done

File: BUFR_TableD_en_10.csv (hunk 5 above)
[the remark this answers is not in the document]
Comment (antoinemerle) [reply BUFR_TableD_en_10.csv]: same as previous comment

File: BUFR_TableD_en_10.csv (hunk 6 above)
[the remark this answers is not in the document]
Comment (antoinemerle) [reply BUFR_TableD_en_10.csv:2034]: in the end I put : Repeat 861 times the Main eigenvectors of the sensitivity matrix

**Comment by antoinemerle** [comment]
Hi @amilan17 ,

I implemented all the comment's you've made.
Really glad that you made this review and gave me feedback. I am taking in account all the notes and advices you gave me.

Please do not hesitate to check again for consistency the full sequence.

On my side I will review and go through my proposal to implement the following across the sequences.
replacing the
```txt
Set to "something" with 9
```
by 
```txt
=9 "Something"
```

**Comment by amilan17** [comment]
@antoinemerle thank you for your updates. I made some copyedit changes that I'd appreciate you taking a quick look at. 89073a1

**Comment by antoinemerle** [comment]
HI @amilan17 , Thanks a million time for your review.
I am really sorry that you spent so much time to edit my PR....

I think we (the group) can try to : 
- have a tiny `pdf` guideline for anyone pushing data in the branch in order to know the correct "nomenclature" to follow to respect the WMO one.
- have a simple `pdf` guide on how to use github and specific branch.

On my side, I am going to implement all the things we saw together in my SW that is converting my `csv` table in the WMO ones...

Thanks again

I review your changes, and it is fine according to me. Thanks a million for chaning the `SO2` to `sulphur dioxide` in the title.

**Pull request merged**